### PR TITLE
Restrict callables to closures in filters

### PR DESCRIPTION
**Pull request by jderusse**
When using filter in sandboxed mode, this PR requires the use of closure.

### CODE REVIEW

File: src/Extension/CoreExtension.php [hunk 1]
```diff
@@ -1693,8 +1693,12 @@ function twig_array_batch($items, $size, $fill = null, $preserveKeys = true)
     return $result;
 }
 
-function twig_array_filter($array, $arrow)
+function twig_array_filter(Environment $env, $array, $arrow)
 {
+    if ($env->getExtension('\Twig\Extension\SandboxExtension')->isSandboxed() && !$arrow instanceof Closure) {
```
Comment (stof) [inline src/Extension/CoreExtension.php]: the leading `\` should be removed, to go through the fast path where the exact class name is provided (`get_class()` in PHP does not return a leading namespace separator)

Comment (nicolas-grekas) [inline src/Extension/CoreExtension.php]: I would suggest swapping both operands.
Also, can't the extension be missing?

**Comment by nicolas-grekas** [comment]
don't miss updating the changelog

File: src/Extension/CoreExtension.php (hunk 1 above)
Comment (stof) [reply src/Extension/CoreExtension.php]: hmm, indeed. We might not have the SandboxExtension registered.

Comment (jderusse) [reply src/Extension/CoreExtension.php]: I added it for consistency with all other calls to `getExtension` and `hasExtension` in the project.

The performance impact is negligible, as the first instruction of getExtension is `$class = ltrim($class, '\\');`

Comment (stof) [reply src/Extension/CoreExtension.php]: hmm, I forgot we un-conditionnally trim the char, without first checking for the key as is.

Comment (jderusse) [reply src/Extension/CoreExtension.php]: good catch! fixed

**Review by nicolas-grekas — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @jderusse.

**Pull request merged**